gdk/common/utils.py

```diff
@@ -165,14 +165,37 @@ def parse_json_error(err):
     msg = err.msg
 
     for err_msg, causes in syntax_error_message.JSON_LIBRARY_ERROR_MESSAGES.items():
-        if msg in err_msg:
+        if err_msg in msg:
             for cause in causes:
                 logging.info("\t " + cause)
             break
 
     logging.info("If none of the above is the cause, please review the overall JSON syntax and resolve any issues.")
 
 
+def parse_yaml_error(err):
+    """
+    Log and display a YAML syntax error message.
+
+    Parameters
+    ----------
+    err : yaml.YAMLError
+        The YAML syntax error.
+
+    """
+    logging.error(f"{err}")
+    logging.info("This might be caused by one of the following reasons: ")
+    msg = err.problem
+
+    for err_msg, causes in syntax_error_message.YAML_LIBRARY_ERROR_MESSAGES.items():
```

Comment:
Have you checked if there are (inner) methods of the `yaml` library that we can use to get the `msg` that looks similar to the ones in the `YAML_LIBRARY_ERROR_MESSAGES`?

Reply:
I've looked into it, and it seems that the YAML library doesn't provide inner methods with error messages that can be accessed publicly. Additionally, there doesn't appear to be a clear documentation that lists all possible error messages.